From 3be3d38518a312610dedecdad89c8cfa050d34e6 Mon Sep 17 00:00:00 2001
From: Boovan Rajendran <Boovan.R@windriver.com>
Date: Wed, 22 Oct 2025 04:28:07 -0400
Subject: [PATCH] Revert "kubeadm: use new etcd /livez and /readyz endpoints"

This reverts commit eeac2dda7091c1f1ea100e7637291bba2345bb0b and
partially reverts commit a04cc020f1b3992fcf185ef2b1bf12c55815fb5a
to install K8s 1.34.1 with the existing etcd version 3.4. This patch
needs to be removed after upgrading the etcd version to >= 3.5.21

Signed-off-by: Boovan Rajendran <Boovan.R@windriver.com>
---
 cmd/kubeadm/app/constants/constants.go      |  2 +-
 cmd/kubeadm/app/constants/constants_test.go | 12 ++++--------
 cmd/kubeadm/app/phases/etcd/local.go        |  9 +++------
 cmd/kubeadm/app/phases/etcd/local_test.go   | 13 ++-----------
 4 files changed, 10 insertions(+), 26 deletions(-)

diff --git a/cmd/kubeadm/app/constants/constants.go b/cmd/kubeadm/app/constants/constants.go
index 21e49e24e6e..ed2528fb75b 100644
--- a/cmd/kubeadm/app/constants/constants.go
+++ b/cmd/kubeadm/app/constants/constants.go
@@ -323,7 +323,7 @@ const (
 	KubeletHealthzPort = 10248
 
 	// MinExternalEtcdVersion indicates minimum external etcd version which kubeadm supports
-	MinExternalEtcdVersion = "3.5.21-0"
+	MinExternalEtcdVersion = "3.4.13-4"
 
 	// DefaultEtcdVersion indicates the default etcd version that kubeadm uses
 	DefaultEtcdVersion = "3.6.4-0"
diff --git a/cmd/kubeadm/app/constants/constants_test.go b/cmd/kubeadm/app/constants/constants_test.go
index 779ec5fda9a..5a46ee20b0c 100644
--- a/cmd/kubeadm/app/constants/constants_test.go
+++ b/cmd/kubeadm/app/constants/constants_test.go
@@ -98,15 +98,11 @@ func TestGetStaticPodFilepath(t *testing.T) {
 	}
 }
 
-func TestEtcdSupportedVersionLength(t *testing.T) {
-	const max = 4
-	if len(SupportedEtcdVersion) > max {
-		t.Fatalf("SupportedEtcdVersion must not include more than %d versions", max)
-	}
-}
-
 func TestEtcdSupportedVersion(t *testing.T) {
 	var supportedEtcdVersion = map[uint8]string{
+		13: "3.2.24",
+		14: "3.3.10",
+		15: "3.3.10",
 		16: "3.3.17-0",
 		17: "3.4.3-0",
 		18: "3.4.3-0",
@@ -125,7 +121,7 @@ func TestEtcdSupportedVersion(t *testing.T) {
 		},
 		{
 			kubernetesVersion: "1.10.1",
-			expectedVersion:   version.MustParseSemantic("3.3.17-0"),
+			expectedVersion:   version.MustParseSemantic("3.2.24"),
 			expectedWarning:   true,
 			expectedError:     false,
 		},
diff --git a/cmd/kubeadm/app/phases/etcd/local.go b/cmd/kubeadm/app/phases/etcd/local.go
index b3728128a79..9c528deea84 100644
--- a/cmd/kubeadm/app/phases/etcd/local.go
+++ b/cmd/kubeadm/app/phases/etcd/local.go
@@ -219,12 +219,9 @@ func GetEtcdPodSpec(cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.A
 					v1.ResourceMemory: resource.MustParse("100Mi"),
 				},
 			},
-			// The etcd probe endpoints are explained here:
-			// https://github.com/kubernetes/kubeadm/issues/3039
-			LivenessProbe:  staticpodutil.LivenessProbe(probeHostname, "/livez", kubeadmconstants.ProbePort, probeScheme),
-			ReadinessProbe: staticpodutil.ReadinessProbe(probeHostname, "/readyz", kubeadmconstants.ProbePort, probeScheme),
-			StartupProbe:   staticpodutil.StartupProbe(probeHostname, "/readyz", kubeadmconstants.ProbePort, probeScheme, componentHealthCheckTimeout),
-			Env:            kubeadmutil.MergeKubeadmEnvVars(cfg.Etcd.Local.ExtraEnvs),
+			LivenessProbe: staticpodutil.LivenessProbe(probeHostname, "/health?exclude=NOSPACE&serializable=true", kubeadmconstants.ProbePort, probeScheme),
+			StartupProbe:  staticpodutil.StartupProbe(probeHostname, "/health?serializable=false", kubeadmconstants.ProbePort, probeScheme, componentHealthCheckTimeout),
+			Env:           kubeadmutil.MergeKubeadmEnvVars(cfg.Etcd.Local.ExtraEnvs),
 			Ports: []v1.ContainerPort{
 				{
 					Name:          kubeadmconstants.ProbePort,
diff --git a/cmd/kubeadm/app/phases/etcd/local_test.go b/cmd/kubeadm/app/phases/etcd/local_test.go
index 16cd66ec948..1fa026790e8 100644
--- a/cmd/kubeadm/app/phases/etcd/local_test.go
+++ b/cmd/kubeadm/app/phases/etcd/local_test.go
@@ -128,7 +128,7 @@ spec:
       failureThreshold: 8
       httpGet:
         host: 127.0.0.1
-        path: /livez
+        path: /health?exclude=NOSPACE&serializable=true
         port: probe-port
         scheme: HTTP
       initialDelaySeconds: 10
@@ -139,15 +139,6 @@ spec:
     - containerPort: 2381
       name: probe-port
       protocol: TCP
-    readinessProbe:
-      failureThreshold: 3
-      httpGet:
-        host: 127.0.0.1
-        path: /readyz
-        port: probe-port
-        scheme: HTTP
-      periodSeconds: 1
-      timeoutSeconds: 15
     resources:
       requests:
         cpu: 100m
@@ -156,7 +147,7 @@ spec:
       failureThreshold: 24
       httpGet:
         host: 127.0.0.1
-        path: /readyz
+        path: /health?serializable=false
         port: probe-port
         scheme: HTTP
       initialDelaySeconds: 10
-- 
2.34.1

